modules/lambdas/main.tf

```diff
@@ -1,83 +1,154 @@
+## Terraform Requirements
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.5.0"
+    }
+  }
+}
+
+
+## AWS Provider Settings
+provider "aws" {
+  region  = var.region
+  profile = var.aws_profile
+}
 
+
+## Local Variables
 locals {
-  default_tags = {
-    Deployment = var.prefix
-  }
+  tags         = merge(var.tags, { Deployment = var.prefix })
+  orca_buckets = [for k, v in var.buckets : v.name if v.type == "orca"]
 }
 
+
+## Referenced Modules
+# lambda_security_group - Security Groups module reference
+# ------------------------------------------------------------------------------
 module "lambda_security_group" {
   source = "../security_groups"
-  tags = var.tags
-  vpc_id = var.vpc_id
-  prefix = var.prefix
+  ## --------------------------
+  ## Cumulus Variables
+  ## --------------------------
+  ## REQUIRED
+  aws_profile = var.aws_profile
+  prefix      = var.prefix
+  vpc_id      = var.vpc_id
+  ## OPTIONAL
+  region = var.region
+  tags   = local.tags
+  ## --------------------------
+  ## ORCA Variables
+  ## --------------------------
+  ## OPTIONAL
+  database_port = var.database_port
 }
 
+# restore_object_arn - IAM module reference
+# ------------------------------------------------------------------------------
 module "restore_object_arn" {
   source = "../iam"
-  buckets = var.buckets
-  prefix = var.prefix
+  ## --------------------------
+  ## Cumulus Variables
+  ## --------------------------
+  ## REQUIRED
+  aws_profile              = var.aws_profile
+  buckets                  = var.buckets
   permissions_boundary_arn = var.permissions_boundary_arn
+  prefix                   = var.prefix
+  ## OPTIONAL
+  region = var.region
+  tags   = local.tags
+  ## --------------------------
+  ## ORCA Variables
+  ## --------------------------
+  ## OPTIONAL
+  orca_recovery_buckets = var.orca_recovery_buckets
 }
 
-// You don't need to specify orca version if you used source_code_hash
-resource "aws_lambda_function" "db_deploy" {
-  filename      = "${path.module}/../../tasks/db_deploy/db_deploy.zip"
-  function_name = "${var.prefix}_db_deploy"
-  source_code_hash = filemd5("${path.module}/../../tasks/db_deploy/db_deploy.zip")
+
+## =============================================================================
+## Ingest Lambdas Definitions and Resources
+## =============================================================================
+
+# copy_to_glacier - Copies files to the ORCA S3 Glacier bucket
+# ==============================================================================
+resource "aws_lambda_function" "copy_to_glacier" {
+  ## REQUIRED
+  function_name = "${var.prefix}_copy_to_glacier"
   role          = module.restore_object_arn.restore_object_role_arn
-  handler       = "db_deploy.handler"
-  runtime       = "python3.7"
```

Comment:
I get the following error while deploying without these two parameters:

"Error: **handler** and **runtime** must be set when PackageType is Zip"

Wondering if these two parameters should stay at ## REQUIRED parameter section in case of .zip package files

Reply:
I based required and optional for the Lambda off of the Terraform documentation https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#argument-reference

I guess I would rather keep them under optional to keep it consistent since we may move to other methods in the future. Otherwise, I would move almost everything under required since I would expect us to use most of the fields provided with only environment being optional between lambdas. Thoughts?

Reply:
yes sounds good to me. We can keep them as it is now.